### ISSUE TITLE
(🐞) `FURB118` (reimplemented-operator) false positive when reference in the item position

### ISSUE THREAD

**Issue by KotlinIsland** [astral-sh/ruff#10898]
```py
b = 1
def f(a):  # Use `operator.itemgetter(b)` instead of defining a function
    return a[b]
b = 2
f([1, 2, 3])  # 3
```
[playground](https://play.ruff.rs/d230918e-ada7-440b-ad19-dc378656ed86)
but if we follow that direction:
```py
b = 1
f = operator.itemgetter(b)
b = 2
f([1, 2, 3])  # 2
```

# Original issue:
```py
class A:
    def __init__(self):
        self.foo = 1
    
    @property
    def f(self) -> object:  # Use `operator.itemgetter(self.foo)` instead of defining a function
        return self[self.foo]
```
[playground](https://play.ruff.rs/d230918e-ada7-440b-ad19-dc378656ed86)

This function references an input in the value position, making it impossible to rewrite as an attribute
```py
class A:
    def __init__(self):
        self.foo = 1

    f = property(operator.itemgetter(self.foo))
``` 

And if we define it in the `__init__` then it would always use the intial value of `foo`:
```py

class A:
    def __init__(self):
        self.foo = 1
        self.f = property(operator.itemgetter(self.foo))
```

**Comment by MichaReiser** [comment]
Yeah, that's certainly not safe. We should either mark the fix as unsafe or "properly" fix (although that would likely require to replace all call-sites of `f` which I don't think we can do easily)

**Comment by KotlinIsland** [comment]
Or detect if there is a reference in the item then don't try to fix it?

**Comment by DetachHead** [comment]
shouldn't it just not even report the error at all in that case, since it means there's no way to properly convert it to an `itemgetter`

**Comment by KotlinIsland** [comment]
Could be a constant
